### ISSUE TITLE
`empty_enum_arguments` false positive for class function

### ISSUE THREAD

**Issue by ecnepsnai** [realm/SwiftLint#4256]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The following code incorrectly triggers the `empty_enum_arguments` rule

```swift
class Example {
    func test() {
        if !MFMailComposeViewController.canSendMail() {
            //
        }
    }
}
```

Apple docs: https://developer.apple.com/documentation/messageui/mfmailcomposeviewcontroller/1616879-cansendmail

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files in current working directory
Linting 'BlahApp.swift' (1/2)
Linting 'ContentView.swift' (2/2)
ContentView.swift:13:52: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated values if they are not used. (empty_enum_arguments)
Done linting! Found 1 violation, 0 serious in 2 files.
```

### Environment

* 0.42.0
* Homebrew
* No configuration file

**Comment by jpsim** [comment]
0.42.0 is a very old version, almost 2 years old. Can you please try with 0.49.1 to see if this has since been resolved? If not, can you try with the latest commit on `main`?

Thanks!

**Comment by ecnepsnai** [comment]
Apologies, I had thought I updated swiftlint before filing this bug but it appears homebrew didn't work. I reinstalled and the false positive went away.
